ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx

```diff
@@ -170,14 +163,22 @@ export function ConfirmAlertModal({
       onClose={onClose}
       customTitle={t('confirmAlertModalTitle')}
       customDetails={
-        <ConfirmDetails onAlertLinkClick={handleOpenMultipleAlertModal} />
+        selectedAlert?.provider === SecurityProvider.Blockaid ? (
+          SecurityProvider.Blockaid
+        ) : (
+          <ConfirmDetails onAlertLinkClick={handleOpenMultipleAlertModal} />
+        )
       }
       customAcknowledgeCheckbox={
         <AcknowledgeCheckboxBase
           selectedAlert={selectedAlert}
           isConfirmed={confirmCheckbox}
           onCheckboxClick={handleConfirmCheckbox}
-          label={t('confirmAlertModalAcknowledge')}
+          label={
+            SecurityProvider.Blockaid
```

Comment:
As mentioned before, we want the alert system to be generic and decoupled from domain. If we add clauses like this, we're increasing that coupling and the level of complexity given the additional unique behaviours.

If we need custom text here for some alerts, could we keep it generic and add a suitable optional property to the `Alert` type itself?

Reply:
I totally agree on this point. My understanding was that how blockaid alerts work currently is temporary solution we arrived on to un-block alerts for signatures. Eventually it should be done in generic way.

If that is not the case definitely we should do it in generic way.

Reply:
If it's only temporary, then that makes total sense but should we create a ticket to log the future refactor to make it more generic?